tests/manage/pv_services/pvc_snapshot/test_restore_with_different_volume_mode.py

```diff
@@ -0,0 +1,434 @@
+import logging
+import pytest
+from itertools import cycle
+
+from ocs_ci.ocs import constants, node
+from ocs_ci.framework.testlib import (
+    skipif_ocs_version,
+    ManageTest,
+    tier2,
+    polarion_id,
+    skipif_ocp_version,
+)
+from ocs_ci.helpers import helpers
+
+log = logging.getLogger(__name__)
+
+
+@tier2
+@skipif_ocs_version("<4.6")
+@skipif_ocp_version("<4.6")
+@polarion_id("OCS-2540")
+class TestSnapshotRestoreWithDifferentVolumeMode(ManageTest):
+    """
+    Tests to verify RBD PVC snapshot restore with volume mode - access mode combination
+    different from that of the parent PVC
+
+    """
+
+    @pytest.fixture(autouse=True)
+    def setup(
+        self,
+        project_factory,
+        snapshot_restore_factory,
+        create_pvcs_and_pods,
+        pod_factory,
+    ):
+        """
+        Create RBD PVCs and pods
+
+        """
+        self.pvcs, self.pods = create_pvcs_and_pods(
+            pvc_size=3,
+            pods_for_rwx=1,
+            access_modes_rbd=[
+                "ReadWriteOnce",
+                "ReadWriteMany-Block",
+                "ReadWriteOnce-Block",
+            ],
+            num_of_rbd_pvc=3,
+            num_of_cephfs_pvc=0,
+        )
+
+    def test_snapshot_restore_with_different_volume_mode(
+        self, pod_factory, snapshot_factory, snapshot_restore_factory
+    ):
+        """
+        Restore snapshot with volume mode - access mode combination different than the parent PVC
+
+        """
+        file_name = "fio_test"
```

Comment:
Can there be some conflict in file name when running the test for second time?
The same for file_name_new.

Should we add some random characters to the file name?

Reply:
Entirely new PVCs are pods ill be created when the test case is running for the second time. So no conflicts will arise.